src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -493,6 +493,7 @@ export class MatAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
         switchMap(() => {
           this._resetActiveItem();
           this.autocomplete._setVisibility();
+          this._changeDetectorRef.detectChanges();
```

Comment:
I see that it's already set up this way, but I just noticed that it's weird for this `switchMap` to have side effects. No action necessary here, but something to keep in mind.

Reply:
@crisbeto @jelbourn As we only need the `detectChanges` when `firstStable` fires (`optionChanges` should not need it, because it is resulting out of an user action) we could limit it to only the `firstStable` event. Maybe like so:

```typescript
return merge(firstStable, optionChanges)
  .pipe(
    // create a new stream of panelClosingActions, replacing any previous streams
    // that were created, and flatten it so our stream only emits closing events...
    switchMap(() => {
      this._resetActiveItem();
      this.autocomplete._setVisibility();

      if (this.panelOpen) {
        this._overlayRef!.updatePosition();
      }

      // Only detectChanges when firstStable fires
      if (!optionChange) {
        this._changeDetectorRef.detectChanges();
      }

      return this.panelClosingActions;
    }),
    // ...
```